db/migrate/20180312080251_change_digests_limit.rb

```diff
@@ -0,0 +1,11 @@
+class ChangeDigestsLimit < ActiveRecord::Migration[5.1]
+  def up
+    change_column :messages, :digest, :string, :limit => 40
+    change_column :sources, :digest, :string, :limit => 40
+  end
+
+  def down
```

Comment:
This was merged quite quickly, but this down migration will lead to inconsistent state relative to what would happen when migrating up to one step earlier with the change to the other migrations. 
Why did the earlier migrations need to change? is it because mysql doesn't support indexing 255 char long strings with unicode? if so, the down migration should be empty, if not, the other migrations shouldn't be changed.

Reply:
see https://github.com/theforeman/foreman/pull/5295#issuecomment-372223679 for why it would fail with utf8 charset, this would block all jenkins jobs since we plan to enable utf there

new migration is there for existing users that decide to start using utf8mb4 encoding from now on (updating connection parameters to fully use utf), the old migration change is necessary for new users or CI, I couldn't find cleaner way, since the new migration applies only for existing users, they already have a limit 255 so down for them means 255

given the fact how often migrating down is needed and how many migrations would fail with irreversible migration, I wouldn't be too concerned about this